lib/saml_idp/assertion_builder.rb

```diff
@@ -58,7 +58,7 @@ def fresh
                   end
               end
             end
-          end
+          end unless config.attributes.nil? || config.attributes.empty?
```

Comment:
Could you put this around the situation we're protecting against rather than at the end here?